// ngx-star-rating/__ivy_ngcc__/fesm2015/ngx-star-rating.js
import { ɵɵdefineInjectable, Injectable, Component, forwardRef, Input, ViewChildren, NgModule } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { CommonModule } from '@angular/common';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

const _c0 = ["ngxCheckbox"];
function NgxStarRatingComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    const _r5 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 2, 3);
    ɵngcc0.ɵɵlistener("click", function NgxStarRatingComponent_ng_template_1_Template_input_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r5); const star_r1 = ctx.$implicit; const ctx_r4 = ɵngcc0.ɵɵnextContext(); return ctx_r4.rate(star_r1); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(2, "label", 4);
} if (rf & 2) {
    const star_r1 = ctx.$implicit;
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMapInterpolate1("star star-", star_r1, "");
    ɵngcc0.ɵɵpropertyInterpolate2("id", "star-", star_r1, "-", ctx_r0.id, "");
    ɵngcc0.ɵɵproperty("checked", ctx_r0.value == star_r1)("disabled", ctx_r0.disabled);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵclassMapInterpolate1("star star-", star_r1, "");
    ɵngcc0.ɵɵpropertyInterpolate2("for", "star-", star_r1, "-", ctx_r0.id, "");
} }
class NgxStarRatingService {
    constructor() { }
}
NgxStarRatingService.ɵfac = function NgxStarRatingService_Factory(t) { return new (t || NgxStarRatingService)(); };
NgxStarRatingService.ɵprov = ɵɵdefineInjectable({ factory: function NgxStarRatingService_Factory() { return new NgxStarRatingService(); }, token: NgxStarRatingService, providedIn: "root" });
NgxStarRatingService.ctorParameters = () => [];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxStarRatingService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();

class NgxStarRatingComponent {
    constructor() {
        this.stars = [5, 4, 3, 2, 1];
        this.propagateChange = (_) => { };
        if (!this.disabled) {
            this.disabled = false;
        }
    }
    ngOnInit() { }
    rate(rate) {
        if (!this.disabled) {
            this.propagateChange(rate);
        }
    }
    writeValue(value) {
        if (this.ngxCheckbox && value === null) {
            this.ngxCheckbox.forEach((checkbox) => {
                checkbox.nativeElement.checked = false;
            });
        }
        this.value = value;
    }
    registerOnChange(fn) {
        this.propagateChange = fn;
    }
    registerOnTouched(fn) { }
}
NgxStarRatingComponent.ɵfac = function NgxStarRatingComponent_Factory(t) { return new (t || NgxStarRatingComponent)(); };
NgxStarRatingComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NgxStarRatingComponent, selectors: [["ngx-star-rating"]], viewQuery: function NgxStarRatingComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, 1);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.ngxCheckbox = _t);
    } }, inputs: { disabled: "disabled", id: "id" }, features: [ɵngcc0.ɵɵProvidersFeature([
            {
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(() => NgxStarRatingComponent),
                multi: true
            }
        ])], decls: 2, vars: 1, consts: [["action", ""], ["ngFor", "", 3, "ngForOf"], ["type", "radio", "name", "star", 3, "id", "checked", "disabled", "click"], ["ngxCheckbox", ""], [3, "for"]], template: function NgxStarRatingComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "form", 0);
        ɵngcc0.ɵɵtemplate(1, NgxStarRatingComponent_ng_template_1_Template, 3, 12, "ng-template", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.stars);
    } }, directives: [ɵngcc1.NgForOf], styles: ["@charset \"UTF-8\";*[_ngcontent-%COMP%]{font-family:roboto;margin:0;padding:0}body[_ngcontent-%COMP%]{background:#000}.cont[_ngcontent-%COMP%]{background:#111;border:thin solid #444;border-radius:5px;color:#eee;margin:4% auto;max-width:350px;overflow:hidden;padding:30px 0;text-align:center;width:93%}hr[_ngcontent-%COMP%]{border:none;border-bottom:thin solid hsla(0,0%,100%,.1);margin:20px}div.title[_ngcontent-%COMP%]{font-size:2em}h1[_ngcontent-%COMP%]   span[_ngcontent-%COMP%]{color:#fd4;font-weight:300}div.stars[_ngcontent-%COMP%]{display:inline-block;width:270px}input.star[_ngcontent-%COMP%]{display:none}label.star[_ngcontent-%COMP%]{color:#444;float:right;font-size:36px;padding:10px;transition:all .2s}input.star[_ngcontent-%COMP%]:checked ~ label.star[_ngcontent-%COMP%]:before{color:#fd4;content:\"\uF005\";transition:all .25s}input.star-5[_ngcontent-%COMP%]:checked ~ label.star[_ngcontent-%COMP%]:before{color:#fe7;text-shadow:0 0 20px #952}input.star-1[_ngcontent-%COMP%]:checked ~ label.star[_ngcontent-%COMP%]:before{color:#f62}label.star[_ngcontent-%COMP%]:hover{transform:rotate(-15deg) scale(1.3)}label.star[_ngcontent-%COMP%]:before{content:\"\uF006\";font-family:FontAwesome}"] });
NgxStarRatingComponent.ctorParameters = () => [];
NgxStarRatingComponent.propDecorators = {
    id: [{ type: Input }],
    disabled: [{ type: Input }],
    ngxCheckbox: [{ type: ViewChildren, args: ['ngxCheckbox',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxStarRatingComponent, [{
        type: Component,
        args: [{
                selector: 'ngx-star-rating',
                template: "<form action=\"\">\r\n  <ng-template ngFor let-star [ngForOf]=\"stars\" let-i=\"index\">\r\n    <input #ngxCheckbox class=\"star star-{{ star }}\" id=\"star-{{ star }}-{{ id }}\" type=\"radio\" name=\"star\" (click)=\"rate(star)\"\r\n      [checked]=\"value == star\" [disabled]=\"disabled\" />\r\n    <label class=\"star star-{{ star }}\" for=\"star-{{ star }}-{{ id }}\"></label>\r\n  </ng-template>\r\n</form>\r\n",
                providers: [
                    {
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef(() => NgxStarRatingComponent),
                        multi: true
                    }
                ],
                styles: ["@charset \"UTF-8\";*{font-family:roboto;margin:0;padding:0}body{background:#000}.cont{background:#111;border:thin solid #444;border-radius:5px;color:#eee;margin:4% auto;max-width:350px;overflow:hidden;padding:30px 0;text-align:center;width:93%}hr{border:none;border-bottom:thin solid hsla(0,0%,100%,.1);margin:20px}div.title{font-size:2em}h1 span{color:#fd4;font-weight:300}div.stars{display:inline-block;width:270px}input.star{display:none}label.star{color:#444;float:right;font-size:36px;padding:10px;transition:all .2s}input.star:checked~label.star:before{color:#fd4;content:\"\uF005\";transition:all .25s}input.star-5:checked~label.star:before{color:#fe7;text-shadow:0 0 20px #952}input.star-1:checked~label.star:before{color:#f62}label.star:hover{transform:rotate(-15deg) scale(1.3)}label.star:before{content:\"\uF006\";font-family:FontAwesome}"]
            }]
    }], function () { return []; }, { disabled: [{
            type: Input
        }], id: [{
            type: Input
        }], ngxCheckbox: [{
            type: ViewChildren,
            args: ['ngxCheckbox']
        }] }); })();

class NgxStarRatingModule {
}
NgxStarRatingModule.ɵfac = function NgxStarRatingModule_Factory(t) { return new (t || NgxStarRatingModule)(); };
NgxStarRatingModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxStarRatingModule });
NgxStarRatingModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxStarRatingModule, { declarations: function () { return [NgxStarRatingComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgxStarRatingComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxStarRatingModule, [{
        type: NgModule,
        args: [{
                declarations: [NgxStarRatingComponent],
                imports: [
                    CommonModule
                ],
                exports: [NgxStarRatingComponent]
            }]
    }], null, null); })();

/*
 * Public API Surface of ngx-star-rating
 */

/**
 * Generated bundle index. Do not edit.
 */

export { NgxStarRatingComponent, NgxStarRatingModule, NgxStarRatingService };
