// ngx-star-rating/bundles/ngx-star-rating.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/forms'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ngx-star-rating', ['exports', '@angular/core', '@angular/forms', '@angular/common'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['ngx-star-rating'] = {}, global.ng.core, global.ng.forms, global.ng.common));
}(this, (function (exports, i0, forms, common) { 'use strict';

    var NgxStarRatingService = /** @class */ (function () {
        function NgxStarRatingService() {
        }
        return NgxStarRatingService;
    }());
    NgxStarRatingService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgxStarRatingService_Factory() { return new NgxStarRatingService(); }, token: NgxStarRatingService, providedIn: "root" });
    NgxStarRatingService.decorators = [
        { type: i0.Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    NgxStarRatingService.ctorParameters = function () { return []; };

    var NgxStarRatingComponent = /** @class */ (function () {
        function NgxStarRatingComponent() {
            this.stars = [5, 4, 3, 2, 1];
            this.propagateChange = function (_) { };
            if (!this.disabled) {
                this.disabled = false;
            }
        }
        NgxStarRatingComponent.prototype.ngOnInit = function () { };
        NgxStarRatingComponent.prototype.rate = function (rate) {
            if (!this.disabled) {
                this.propagateChange(rate);
            }
        };
        NgxStarRatingComponent.prototype.writeValue = function (value) {
            if (this.ngxCheckbox && value === null) {
                this.ngxCheckbox.forEach(function (checkbox) {
                    checkbox.nativeElement.checked = false;
                });
            }
            this.value = value;
        };
        NgxStarRatingComponent.prototype.registerOnChange = function (fn) {
            this.propagateChange = fn;
        };
        NgxStarRatingComponent.prototype.registerOnTouched = function (fn) { };
        return NgxStarRatingComponent;
    }());
    NgxStarRatingComponent.decorators = [
        { type: i0.Component, args: [{
                    selector: 'ngx-star-rating',
                    template: "<form action=\"\">\r\n  <ng-template ngFor let-star [ngForOf]=\"stars\" let-i=\"index\">\r\n    <input #ngxCheckbox class=\"star star-{{ star }}\" id=\"star-{{ star }}-{{ id }}\" type=\"radio\" name=\"star\" (click)=\"rate(star)\"\r\n      [checked]=\"value == star\" [disabled]=\"disabled\" />\r\n    <label class=\"star star-{{ star }}\" for=\"star-{{ star }}-{{ id }}\"></label>\r\n  </ng-template>\r\n</form>\r\n",
                    providers: [
                        {
                            provide: forms.NG_VALUE_ACCESSOR,
                            useExisting: i0.forwardRef(function () { return NgxStarRatingComponent; }),
                            multi: true,
                        }
                    ],
                    styles: ["@charset \"UTF-8\";*{font-family:roboto;margin:0;padding:0}body{background:#000}.cont{background:#111;border:thin solid #444;border-radius:5px;color:#eee;margin:4% auto;max-width:350px;overflow:hidden;padding:30px 0;text-align:center;width:93%}hr{border:none;border-bottom:thin solid hsla(0,0%,100%,.1);margin:20px}div.title{font-size:2em}h1 span{color:#fd4;font-weight:300}div.stars{display:inline-block;width:270px}input.star{display:none}label.star{color:#444;float:right;font-size:36px;padding:10px;transition:all .2s}input.star:checked~label.star:before{color:#fd4;content:\"\uF005\";transition:all .25s}input.star-5:checked~label.star:before{color:#fe7;text-shadow:0 0 20px #952}input.star-1:checked~label.star:before{color:#f62}label.star:hover{transform:rotate(-15deg) scale(1.3)}label.star:before{content:\"\uF006\";font-family:FontAwesome}"]
                },] }
    ];
    NgxStarRatingComponent.ctorParameters = function () { return []; };
    NgxStarRatingComponent.propDecorators = {
        id: [{ type: i0.Input }],
        disabled: [{ type: i0.Input }],
        ngxCheckbox: [{ type: i0.ViewChildren, args: ['ngxCheckbox',] }]
    };

    var NgxStarRatingModule = /** @class */ (function () {
        function NgxStarRatingModule() {
        }
        return NgxStarRatingModule;
    }());
    NgxStarRatingModule.decorators = [
        { type: i0.NgModule, args: [{
                    declarations: [NgxStarRatingComponent],
                    imports: [
                        common.CommonModule
                    ],
                    exports: [NgxStarRatingComponent]
                },] }
    ];

    /*
     * Public API Surface of ngx-star-rating
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.NgxStarRatingComponent = NgxStarRatingComponent;
    exports.NgxStarRatingModule = NgxStarRatingModule;
    exports.NgxStarRatingService = NgxStarRatingService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
